ethereum/contracts/pyth/PythUpgradable.sol

```diff
@@ -65,11 +67,25 @@ contract PythUpgradable is Initializable, OwnableUpgradeable, UUPSUpgradeable, P
         PythSetters.setValidTimePeriodSeconds(newValidTimePeriodSeconds);
     }
 
+    // Privileged function to update the governance emitter
+    function updateGovernanceDataSource(uint16 chainId, bytes32 emitter, uint64 sequence) onlyOwner public {
+        PythInternalStructs.DataSource memory ds = PythInternalStructs.DataSource(chainId, emitter);
+        PythSetters.setGovernanceDataSource(ds);
+        PythSetters.setLastExecutedGovernanceSequence(sequence);
+    }
+
     /// Ensures the contract cannot be uninitialized and taken over.
     /// @custom:oz-upgrades-unsafe-allow constructor
     constructor() initializer {}
 
     // Only allow the owner to upgrade the proxy to a new implementation.
     function _authorizeUpgrade(address) internal override onlyOwner {}
 
+    // Execute a UpgradeContract governance message
+    function upgradeUpgradableContract(UpgradeContractPayload memory payload) override internal {
+        address oldImplementation = _getImplementation();
+        _upgradeToAndCallUUPS(payload.newImplementation, new bytes(0), false);
```

Comment:
does this do any sort of sanity checking to ensure that `payload.newImplementation` points to a new contract? I.e., is there any protection against us accidentally bricking the contract by passing in the wrong value for `newImplementation`?

Reply:
One thing we could do is add a public `isPyth` function, which we call here to check that the new implementation is actually the Pyth contract.